tests/inputFields.spec.ts

```diff
@@ -0,0 +1,80 @@
+import { test, expect } from '@playwright/test';
+import exp from 'constants';
+
+test.beforeEach(async({page}) => {
+    await page.goto('https://petclinic.bondaracademy.com/welcome')
+})
+
+test.describe('Pet types', () => {
+    test.beforeEach(async({page}) => {
+        await page.getByRole('link', {name: "Pet Types"}).click() //1. Select Pet Types menu
+    })
+
+    test('Update pet type', async({page}) => {
+        const petTypesTableName = page.getByRole('heading', {name: "Pet Types"})
+        await expect(petTypesTableName).toHaveText('Pet Types')  //2. assertion of Pet Types text displayed
+
+        await page.getByRole('button', {name: "Edit"}).first().click()  //3. click on edit button for cat
+        await expect(page.getByRole('heading', {name: "Edit Pet Type"})).toBeVisible() //4. assertion of Edit Pet Type header
+
+        const editPetTypeName = page.locator('#name')
+        await editPetTypeName.click()
+        await editPetTypeName.clear()
+        await editPetTypeName.fill('rabbit') //5. change pet name to rabbit and click update
+        const updatedPetTypeNameValue = await editPetTypeName.inputValue()
+        expect(updatedPetTypeNameValue).toEqual('rabbit')
+        const updateButton = page.getByRole('button', {name: "Update"})
+        await updateButton.click()
+
+        const petTypesListName = page.locator('tr input').nth(0)
+        await expect(petTypesListName).toHaveValue('rabbit') //6. assertion for first pet in list is rabbit
+
+        await page.getByRole('button', {name: "Edit"}).first().click() //7. click edit button for rabbit
```

Comment:
duplication of the locator from the line 17.

Reply:
Should I be saving the Edit button into a constant initially and then just reusing the constant?

Reply:
Correct.
If you have just one copy/paste, it can be acceptable depends on the context. But if 3 times, need to save locator to constant 100%.

Reply:
I have saved the edit button to a const now but it is in each test. Can I save the constant in the test describe and call it in each test?

Reply:
Did you mean in beforeEach hook? Unfortunately you can't.

Reply:
Yes, that's what I meant.